com.minres.coredsl/src/com/minres/coredsl/analysis/AnalysisContext.xtend

```diff
@@ -0,0 +1,20 @@
+package com.minres.coredsl.analysis
+
+import org.eclipse.xtext.validation.ValidationMessageAcceptor
+import java.util.Map
+import com.minres.coredsl.coreDsl.ISA
+import java.util.HashMap
+import java.util.Set
+
+class AnalysisContext {
+	
+	public val ValidationMessageAcceptor acceptor;
+	
+	/** Maps each ISA to a set of names declared within it and its parents */
+	public val Map<ISA, Set<String>> exposedDeclarations = new HashMap();
+	public val Map<ISA, ElaborationContext> elaborationResults = new HashMap();
+	
+	new(ValidationMessageAcceptor acceptor) {
+		this.acceptor = acceptor;
+	}
+}
```

Comment:
Nit: Check that your new files have newlines at the end, please.

Reply:
Will do. Could we maybe add an .editorconfig file to do these things for us? #10 seems relevant. I have however noticed that the formatter settings for .xtend files are very minimal and are missing most of the options available for Java code, so I'm not even sure if that is supported.

Reply:
Yeah, I'll look into doing #10 after this PR lands. Good point, an `.editorconfig` would be better than Eclipse-specific configuration.

> I have however noticed that the formatter settings for .xtend files are very minimal and are missing most of the options available for Java code, so I'm not even sure if that is supported.

Ah, bummer!